Composer/packages/extensions/visual-designer/src/editors/ObiEditor.tsx

```diff
@@ -89,16 +87,19 @@ export const ObiEditor: FC<ObiEditorProps> = ({
         if (eventData.$type === 'PASTE') {
           handler = e => {
             // TODO: clean this along with node deletion.
-            const copyLgTemplateToNewNode = async (lgTemplateName: string, newNodeId: string) => {
-              const matches = /\[(bfd\w+-(\d+))\]/.exec(lgTemplateName);
-              if (Array.isArray(matches) && matches.length === 3) {
-                const originLgId = matches[1];
-                const originNodeId = matches[2];
-                const newLgId = originLgId.replace(originNodeId, newNodeId);
-                await copyLgTemplate('common', originLgId, newLgId);
-                return `[${newLgId}]`;
-              }
-              return lgTemplateName;
+            const copyLgTemplateToNewNode = async (input: string, newNodeId: string) => {
+              const lgTemplateRef = LgTemplateRef.parse(input);
+              if (!lgTemplateRef) return input;
+
+              const lgMetadata = LgMetaData.parse(lgTemplateRef.name);
+              if (!lgMetadata) return input;
+
+              lgMetadata.designerId = newNodeId;
+              const newLgName = lgMetadata.toLgTemplateName();
+              const newLgTemplateRefString = lgMetadata.toLgTemplateRefString();
+
+              await copyLgTemplate('common', lgTemplateRef.name, newLgName);
```

Comment:
`copyLgTemplate` will be defined as `copyLgText` in my following PR. Targets to hide LG concepts from visual editor, will inject a synthesized lg api from Shell level.